app/src/main/java/com/google/android/ground/ui/datacollection/tasks/polygon/DrawAreaTaskFragment.kt

```diff
@@ -104,6 +107,22 @@ class DrawAreaTaskFragment @Inject constructor() : AbstractTaskFragment<DrawArea
     if (isVisible && !viewModel.instructionsDialogShown) {
       showInstructionsDialog()
     }
+    viewLifecycleOwner.lifecycleScope.launch {
+      viewModel.showSelfIntersectionDialog.collect {
+        renderComposableDialog {
+          val showDialog = remember { mutableStateOf(true) }
+          if (showDialog.value) {
```

Comment:
We don't need this. ConfirmationDialog already maintains an internal state for visiblity